src/main/java/Strategy/ConcreteAutoGameInitStrategy.java

```diff
@@ -6,15 +6,92 @@
 import model.ConnectorType;
 
 import java.util.HashMap;
+import java.util.HashSet;
 
 public class ConcreteAutoGameInitStrategy implements GameInitStrategy {
     @Override
-    public GameBuilder initializeGame(GameBuilder builder, ValidatorFacade validatorFacade) {
+    public GameBuilder initializeGame(GameBuilder gameBuilder, ValidatorFacade validatorFacade) {
+
+        gameBuilder.setLadders(initializeLadders(gameBuilder, validatorFacade));
+        gameBuilder.setSnakes(initializeSnakes(gameBuilder, validatorFacade));
+
+        return gameBuilder;
+    }
+
+    @Override
+    public HashMap<Integer, Connector> initializeSnakes(GameBuilder gameBuilder, ValidatorFacade validatorFacade) {
+
         HashMap<Integer, Connector> snakes = new HashMap<>();
+
+        /*
+        Assuming boardSize = 100
+        Snakes: start-> (2,99) ; end-> (1, start-1)
+        Ladders: start-> (1,99) ; end-> (start+1,100)
+         */
+
+        HashSet<Integer> startPositions = new HashSet<>();
+        HashSet<Integer> endPositions = new HashSet<>();
+
+        int start, end;
+        int snakeMinStart = 2;
+
```

Comment:
I see you have used a Hashset to ensure start and end positions are not repeated, however you can still end up with a snake ladder loop. 

Create instance variable
private HashMap<Integer, Connector> snakes = new HashMap<Integer, Connector>();
private HashMap<Integer, Connector> ladders = new HashMap<Integer, Connector>();

Then in initialize snakes function,

```suggestion
for (int i = 0; i < gameBuilder.getNumSnakes(); i++) {
  start = (int) (Math.random() * (gameBuilder.getBoardSize() - snakeMinStart) + snakeMinStart);
  end = (int) (Math.random() * (start - 1) + 1);
  Connector snake = new Connector(start, end, ConnectorType.SNAKE);
  if (validatorFacade.getFirstValidator().validate(snake, this.snakes, this,ladders)
  { snakes.put(snake.getStart(), snake);
  }
  else
    continue;
```